crates/bootstrap_srv/src/store.rs

```diff
@@ -0,0 +1,193 @@
+//! This is a virtual-memory inspired tempfile store.
+//!
+//! ### Rationale
+//!
+//! - We don't need to persist anything in the store beyond a single
+//!   process invocation, because the infos are going to expire after
+//!   a matter of minutes anyways, and peers will continue to re-publish them.
+//! - We would like a server to be able to store more spaces and infos
+//!   than would reasonably fit in RAM.
+//! - We would like to avoid use of an external database, to reduce
+//!   dependencies and ease deployment and testing.
+//!
+//! ### Implementation
+//!
+//! - Have a pool of tempfiles that can grow to match the worker thread
+//!   count if needed.
+//! - Have the ability to Clone (reopen) these tempfile handles.
+//! - Designate one file handle only per pool entry as writable.
+//! - All other handles will be readonly.
+//! - After writing an entry, return a reference with a readonly file handle
+//!   to the tempfile that was written to, and an offset/length to
+//!   allow readback of exactly the bytes written for that entry.
+//! - Once a tempfile reaches a certain size, drop the write handle and
+//!   open a new tempfile for writing.
+//! - The older read handles will persist the existence of the older tempfiles
+//!   until the last read reference is dropped, at which point the tempfile
+//!   will be cleaned up by the os.
+//! - We can trust these files will cycle through at a rate similar to the
+//!   max expiration time on the infos they contain (30 minutes).
+
+use std::sync::{Arc, Mutex};
+
+/// How large we should allow individual tempfiles to grow.
+/// This is a balance between using up too much disk space
+/// and having too many file handles open.
+///
+/// Start with 10MiB?
+const MAX_PER_TEMPFILE: u64 = 1024 * 1024 * 10;
```

Comment:
Should this also be in the config?

Reply:
The pressure leading to wanting this larger is the inefficiency of having multiple tempfiles open. This lets us store more than 10_000 entries in one file (because infos are often much less than 1024 bytes long). It doesn't feel like we should need this to be much larger.

The pressure leading to wanting this smaller is the constraints of disk space. On a 16 core system (which cloud servers will likely not have anywhere near that) the default production config will have 64 active write tempfiles, giving 64MiB. Even if we have 10 older generations still around, that's only 640 MiB. Do we think servers will ever have that little disk space? Or to look at it from the opposite perspective, let's say the cloud server provides 10GiB of disk space. We can store 10,485,760 agent records at once, even if many of those are defunct, waiting to be cleaned up.

I don't *feel* like dev-ops folks are going to understand the consequences of making changes to this number. Providing the option to may just be confusing, and not practically useful. What do you think?

P.S. Re-reading this post, I'm not sure I even described it without being confusing...